### PR TITLE
add endian conversion support to lualinux

### PR DESCRIPTION
**Pull request by sheharyaar**

### CODE REVIEW

**Review by lneto — COMMENTED** [review]
as we are creating higher level APIs, I believe we don't need to be tied up to the Linux's namespace.. I think [NetBSD's API names](https://man.netbsd.org/le16dec.9) are more sane.. e.g., `be32toh` instead of `be32_to_cpu`. What do you think? Also, aren't `ntoh*` and `hton*` just aliases to `be*_to_cpu` and `cpu_to_be*` respectively?

@sav would be good to hear your thoughts about this as well ;-)

File: lib/lualinux.c [hunk 1]
```diff
@@ -104,6 +105,34 @@ static int lualinux_lookup(lua_State *L)
...
+LUALINUX_NEW_ENDIAN_CONVERTER(htonl, uint32_t);
+LUALINUX_NEW_ENDIAN_CONVERTER(cpu_to_be16, uint32_t);
+LUALINUX_NEW_ENDIAN_CONVERTER(cpu_to_be32, uint32_t);
+LUALINUX_NEW_ENDIAN_CONVERTER(cpu_to_le16, uint32_t);
+LUALINUX_NEW_ENDIAN_CONVERTER(cpu_to_le32, uint32_t);
+LUALINUX_NEW_ENDIAN_CONVERTER(be16_to_cpu, uint32_t);
+LUALINUX_NEW_ENDIAN_CONVERTER(be32_to_cpu, uint32_t);
+LUALINUX_NEW_ENDIAN_CONVERTER(le16_to_cpu, uint32_t);
+LUALINUX_NEW_ENDIAN_CONVERTER(le32_to_cpu, uint32_t);
+
```
Comment (lneto) [inline lib/lualinux.c:128]:
```suggestion
```

File: lib/lualinux.c [hunk 2]
```diff
@@ -104,6 +105,34 @@ static int lualinux_lookup(lua_State *L)
 	return 1;
 }
 
+#define LUALINUX_NEW_ENDIAN_CONVERTER(method, T) \
```
Comment (lneto) [inline lib/lualinux.c:108]:
```suggestion
#define LUALINUX_NEW_BYTESWAPPER(func, T) \
```

"byte swapper" is a bit shorter than "endian converter".. and `method` is a bit misleading as it is not operating on an object..

Comment (lneto) [reply lib/lualinux.c:108]: good work on this template, btw 💪

**Comment by sheharyaar** [comment]
> as we are creating higher level APIs, I believe we don't need to be tied up to the Linux's namespace.. I think [NetBSD's API names](https://man.netbsd.org/le16dec.9) are more sane.. e.g., `be32toh` instead of `be32_to_cpu`. What do you think? Also, aren't `ntoh*` and `hton*` just aliases to `be*_to_cpu` and `cpu_to_be*` respectively?
> 
> @sav would be good to hear your thoughts about this as well ;-)

Yes, `hton*` and `ntoh*` are aliases to `be*_to_cpu`, we can move to it, I used the native names for easy macro.

**Review by lneto — APPROVED** [review]

**Pull request merged**